app/models/govspeak_url_checker.rb

```diff
@@ -4,15 +4,22 @@ def initialize govspeak
   end
 
   def find_broken_urls
-    extract_all_urls_from_govspeak.select do |url|
-      checked_url = CheckedUrl.find_or_initialize_by(url: url)
-      if checked_url.new_record? || checked_url.expired?
-        checked_url.ok = check_url(url)
-        checked_url.save!
-      end
+    urls_to_check = extract_all_urls_from_govspeak.select do |url|
+      url = CheckedUrl.find_or_initialize_by(url: url)
+      url.new_record? || url.expired?
+    end
 
-      !checked_url.ok?
+    broken_urls = Parallel.map(urls_to_check, in_threads: 10) do |u|
+      check_url(u) ? nil : u
```

Comment:
Did you first try changing `#check_url` to perform a `HEAD` request? This is probably piles more efficient.